features/step_definitions/repository_steps.rb

```diff
@@ -163,11 +167,19 @@
 end
 
 When(/^I push some commits to the repository$/) do
-  post repository_notify_push_path(id: @repository.id), {}, {'HTTP_X_GITLAB_EVENT' => 'Push Hook'}
+  request = FactoryGirl.build(:gitlab_webhook_request)
+  request.headers.each do |k, v|
+    header k, v
```

Comment:
@rafamanzo It seems it doesn't work for this particular case. The `post` method being used comes from `Rack::Test::Helpers`, not the Capybara page driver, so setting headers for it like that doesn't help.